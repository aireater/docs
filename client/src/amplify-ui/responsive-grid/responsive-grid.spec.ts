import {AmplifyResponsiveGrid} from "./responsive-grid";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-responsive-grid", () => {
  describe("Component logic", () => {
    let amplifyResponsiveGrid: AmplifyResponsiveGrid;
    beforeEach(() => (amplifyResponsiveGrid = new AmplifyResponsiveGrid()));

    it("should init `gridGap` as `2`", () =>
      expect(amplifyResponsiveGrid.gridGap).toBe(2));
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifyResponsiveGrid],
            html: `<amplify-grid />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
