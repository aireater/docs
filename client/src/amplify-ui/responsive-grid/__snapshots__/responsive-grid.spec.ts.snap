// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-responsive-grid Render logic should render 1`] = `MockHTMLElement {}`;
